client/client-sdk-javascript/test/client-sdk.spec.ts

```diff
@@ -17,37 +18,53 @@ const senderAddress = new Address(publicKey, VIRTUAL_CHAIN_ID, Address.TEST_NETW
 const API_ENDPOINT = "http://localhost:8888";
 const TIMEOUT = 20;
 
+const universalAddress: UniversalAddress = {
+  id: new Buffer(senderAddress.toString()),
+  networkId: Number(Address.TEST_NETWORK_ID),
+  checksum: 0,
+  scheme: 0
+};
+
+const expectedTransaction: Transaction = {
+  header: {
+    version: 0,
+    sender: universalAddress,
+    timestamp: Date.now().toString()
+  },
+  body: {
+    contractAddress: {
+      address: "contractAddress"
+    },
+    payload: "payload"
+  }
+};
+
+const expectedContract: CallContractInput = {
+  sender: universalAddress,
+  contractAddress: {
+    address: "contractAddress"
+  },
+  payload: "call-payload"
+};
+
 describe("A client calls the connector interface with the correct inputs when", async function () {
   let orbsClient: OrbsClient;
-  const connection = stubInterface<PublicApiConnection>();
+  let httpServer: Server;
 
   beforeEach(async () => {
     orbsClient = new OrbsClient(API_ENDPOINT, senderAddress.toString(), TIMEOUT);
+    httpServer = mockHttpServer(expectedTransaction, expectedContract).listen(8888);
   });
 
   it("sendTransaction() is called", async () => {
-    nock(API_ENDPOINT)
-      .post("/public/sendTransaction", (res: any) => {
-        expect(res.transaction.body).to.be.eql({
-          contractAddress: {
-            address: "contractAddress"
-          },
-          payload: "payload"
-        });
-        return true;
-      }).reply(200, { result: "ok" });
-
     expect(await orbsClient.sendTransaction("contractAddress", "payload")).to.be.eql("ok");
   });
 
   it("callContract() is called", async () => {
-    nock(API_ENDPOINT)
-      .post("/public/callContract", (res: any) => {
-        expect(res.contractAddress).to.be.eql({ address: "contractAddress" });
-        expect(res.payload).to.be.eql("call-payload");
-        return true;
-      }).reply(200, { result: 12 });
-
-    expect(await orbsClient.call("contractAddress", "call-payload")).to.be.eql(12);
+    expect(await orbsClient.call("contractAddress", "call-payload")).to.be.eql("some-answer");
```

Comment:
this test isn't symmetrical - how should the reader know why the server is expected to return `"some-answer"`? it's better to make `mockHttpServer` programmable and to set it up in a previous line.
Not urgent for now, but we *will* need it eventually and I think maybe it's better to do it now so that the test is clearer.